src/olympia/reviewers/tests/test_views.py

```diff
@@ -6991,8 +6991,8 @@ def test_compare_basic(self):
             'version_pk': self.version.pk,
             'pk': new_version.pk})
 
-        with self.assertNumQueries(12):
-            # FIXME: check queries and see if there is a need to optimize, 12
+        with self.assertNumQueries(14):
+            # FIXME: check queries and see if there is a need to optimize, 14
```

Comment:
I'm simply increasing this here, as investigating all the potential ways to optimize this is far beyond this issue.

Reply:
I think we should make sure we're not regressing, though. Can you add the right `select_related` to prevent the number of queries from increasing ?

Reply:
No, I can't easily in this case. I need to load the file-validation which is attached via `version.files` which - unless I'm missing something - needs a `prefetch_related` so the best I can think of is something along the lines of

```diff
diff --git a/src/olympia/reviewers/serializers.py b/src/olympia/reviewers/serializers.py
index 382b869008..aa9c1b15c9 100644
--- a/src/olympia/reviewers/serializers.py
+++ b/src/olympia/reviewers/serializers.py
@@ -407,8 +407,13 @@ class FileEntriesDiffSerializer(FileEntriesSerializer):
         parent = self.context['parent_version']
         selected_file = self.get_selected_file(obj)

-        validation_objects = FileValidation.objects.filter(file_id__in=(
-            (obj.pk, parent.current_file.pk)))
+        validation_objects = []
+
+        for fobj in (obj, parent.current_file):
+            try:
+                validation_objects.append(fobj.validation)
+            except FileValidation.DoesNotExist:
+                continue

         for validation in validation_objects:
             data = json.loads(validation.validation)
diff --git a/src/olympia/reviewers/tests/test_views.py b/src/olympia/reviewers/tests/test_views.py
index b2a77fb570..be7b63e592 100644
--- a/src/olympia/reviewers/tests/test_views.py
+++ b/src/olympia/reviewers/tests/test_views.py
@@ -6991,8 +6991,8 @@ class TestReviewAddonVersionCompareViewSet(
             'version_pk': self.version.pk,
             'pk': new_version.pk})

-        with self.assertNumQueries(14):
-            # FIXME: check queries and see if there is a need to optimize, 14
+        with self.assertNumQueries(12):
+            # FIXME: check queries and see if there is a need to optimize, 12
             # seems too high.
             response = self.client.get(self.url + '?file=README.md&lang=en-US')
         assert response.status_code == 200
diff --git a/src/olympia/reviewers/views.py b/src/olympia/reviewers/views.py
index 1a7ec74d2e..deb81db804 100644
--- a/src/olympia/reviewers/views.py
+++ b/src/olympia/reviewers/views.py
@@ -1504,6 +1504,9 @@ class ReviewAddonVersionDraftCommentViewSet(
 class ReviewAddonVersionCompareViewSet(ReviewAddonVersionMixin,
                                        RetrieveModelMixin, GenericViewSet):

+    def filter_queryset(self, qset):
+        return qset.prefetch_related('files__validation')
+
     def retrieve(self, request, *args, **kwargs):
         serializer = AddonCompareVersionSerializer(
             instance=self.get_object(),
```

to ensure the query count is predictable but it doesn't reduce it. I'll try to spend some more time to try to avoid a regression.

Reply:
If it stays at 12 it's fine. We'll look at how to reduce it separately, I'm just concerned about not making things worse for now.

Reply:
Once https://github.com/mozilla/addons-server/pull/13726 is merged the number of queries should stay at 10 even when using the file validation info for the second file in the compare view.